uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/models/instances/leave-request-instance.js

```diff
@@ -53,8 +53,34 @@ define([
          * @return {Promise} Resolved with {Object} Updated Leave request
          */
         update: function (attributes) {
-          return LeaveRequestAPI.sendPOST('LeaveRequest', 'create', _.assign({}, this.attributes(), attributes))
+          var updatedAttributes = _.assign({}, this.attributes(), attributes);
+          return LeaveRequestAPI.update(updatedAttributes);
+        },
+
+        /**
+         * This method is used to create a new leave request
+         *
+         * @param {object} attributes - Values which will be used to create new leave request
+         * @return {Promise} Resolved with {Object} Created Leave request
+         */
+        create: function (attributes) {
```

Comment:
Same for `update`, I don't think `create()` should accept any parameter, it should just send to the api the data already stored in the instance

```js
var instance = LeaveRequest.init({ /* some custom data */ });

instance.create().then(function () {
  instance.id; // the instance has now an id
});
```